modules/gapi/src/backends/cpu/gcpustereo.cpp

```diff
@@ -0,0 +1,49 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2021 Intel Corporation
+
+#include <opencv2/gapi/stereo.hpp>
+#include <opencv2/gapi/cpu/stereo.hpp>
+#include <opencv2/gapi/cpu/gcpukernel.hpp>
+
+#include <opencv2/calib3d.hpp>
+
+GAPI_OCV_KERNEL_ST(GCPUStereo, cv::gapi::ster::GStereo, cv::gapi::ster::StereoSetup)
+{
+    static void setup(const cv::GMatDesc&, const cv::GMatDesc&,
+                      const cv::gapi::ster::OUTPUT_FORMAT, const int&,
+                      std::shared_ptr<cv::gapi::ster::StereoSetup> &stereoSetup,
+                      const cv::GCompileArgs &compileArgs) {
+        auto sSetup = cv::gapi::getCompileArg<cv::gapi::ster::StereoSetup>(compileArgs)
+            .value_or(cv::gapi::ster::StereoSetup{});
+        stereoSetup = std::make_shared<cv::gapi::ster::StereoSetup>(sSetup);
+    }
+
+    static void run(const cv::Mat& left,
+                    const cv::Mat& right,
+                    const cv::gapi::ster::OUTPUT_FORMAT oF,
+                    const int dtype,
+                    cv::Mat& out_disp,
+                    cv::gapi::ster::StereoSetup stereoSetup
+                    )
+    {
+        cv::Mat disp;
+        cv::StereoBM::create(
```

Comment:
You shouldn't create new instance  each time for cv::StereoBM if you have opportunity to use state...

Reply:
Fixed